crypto/keys/src/ed25519.rs

```diff
@@ -333,29 +337,33 @@ impl TryFrom<alloc::vec::Vec<u8>> for Ed25519Private {
 
 #[derive(Debug)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Ed25519Pair(Keypair);
+pub struct Ed25519Pair(SigningKey);
 
 impl Ed25519Pair {
     pub fn private_key(&self) -> Ed25519Private {
-        Ed25519Private::try_from(self.0.secret.as_ref()).expect("Invalid private key in keypair")
+        let secret_key = self.0.to_bytes();
+        Ed25519Private::try_from(secret_key.as_slice()).expect("Invalid private key in keypair")
     }
 
     pub fn public_key(&self) -> Ed25519Public {
-        Ed25519Public(self.0.public)
+        Ed25519Public(self.0.verifying_key())
     }
 }
 
 impl<D: Digest<OutputSize = U64>> DigestSigner<D, Ed25519Signature> for Ed25519Pair {
     fn try_sign_digest(&self, digest: D) -> Result<Ed25519Signature, SignatureError> {
-        let sig = self.0.sign_prehashed(digest, None)?;
+        let sig = self
+            .0
+            .sign_prehashed(digest, None)
+            .map_err(|_e| SignatureError::new())?;
```

Comment:
Suggest implementing `From<Whatever> for SignatureError` instead.

Reply:
Done.